docs/docs/user/migration-guide/v1-v2.md

```diff
@@ -0,0 +1,175 @@
+# Migrate from V1 to V2
+
+## [Derive component type automatically from class name](https://github.com/bakdata/kpops/pull/309)
+
+KPOps automatically inferes the `type` from the class name. Therefore, there is no need to add the `type` anymore. By convention the `type` would be the lower, and kebab cased name of the class.
+
+```diff
+class MyCoolStreamApp(StreamsApp):
+-    type = "my-cool-stream-app"
++    ...
+```
+
+The [ProducerApp](https://github.com/bakdata/kpops/blob/main/kpops/components/streams_bootstrap/producer/producer_app.py#L16) class, which had the `type` `producer`, now through these changes and name convention hast the `type` `producer-app`. This must be addressed in your `pipeline.yaml` or `defaults.yaml`.
```

Comment:
imo just saying: "because of this new convention `producer` has been renamed to `producer-app`" would be enough